tuf/data/roles.go

```diff
@@ -69,6 +69,16 @@ func ValidRole(name string) bool {
 	return false
 }
 
+// IsBaseRole checks if a role name is valid base role
+func IsBaseRole(name string) bool {
+	for _, v := range BaseRoles {
+		if name == v {
+			return true
+		}
+	}
+	return false
+}
+
 // IsDelegation checks if the role is a delegation or a root role
 func IsDelegation(role string) bool {
```

Comment:
Now that we have a specific type for delegated roles, we should remove this and replace with a type check too.